### PR TITLE
Issue #738 - Allow configuring google analytics tracking

### LINKED ISSUE #738: Google analytics integration
It should be possible to enable google analytics tracking. GA ID should be specified in argocd-cm config map and accessible via API

### PR DESCRIPTION
**Pull request by alexmt**
Closes #738

### CODE REVIEW

File: ui/src/app/shared/services/auth-service.ts [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
     public settings(): Promise<AuthSettings> {
         return requests.get('/settings').then((res) => res.body as AuthSettings);
     }
+
+    public getCurrentUserId(): string {
+        const cookies = parse(document.cookie);
+        const token = cookies['argocd.token'];
+        const user: any = token && jwt(token) || null;
+        if (user) {
+            return user.email || user.sub;
```
Comment (alexec) [inline ui/src/app/shared/services/auth-service.ts]: should be anon?

**Review by alexec — APPROVED** [review]
One comment.

File: ui/src/app/shared/services/auth-service.ts (hunk 1 above)
Comment (alexmt) [reply ui/src/app/shared/services/auth-service.ts]: it makes sense. changed to return 'anonymous' is there is no user

Comment (alexec) [reply ui/src/app/shared/services/auth-service.ts]: sorry - I meant to say we should obfuscate the UID before we send it to Google

Comment (alexmt) [reply ui/src/app/shared/services/auth-service.ts]: as we agreed offline, added setting to obfuscate user ids. By default user ids are obfuscated

**Pull request merged**